src/py_ext_control_template/controller.py

```diff
@@ -0,0 +1,411 @@
+## Controller interface for drone toolbox in python
+## by Max Lodel, TU Delft, 2022
+
+import enum
+import numpy as np
+
+import rospy
+from drone_toolbox_ext_control_template.cfg import ControllerConfig
+
+from mavros_msgs.msg import PositionTarget, AttitudeTarget
+from mav_msgs.msg import RollPitchYawrateThrust
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import Pose
+
+from std_srvs.srv import Trigger
+
+from py_ext_control_template.custom_timer import CustomTimer
+
+from tf.transformations import quaternion_from_euler
+
+class ControlMethod(enum.Enum):
+    POS_CTRL = 0
+    POS_YAW_CTRL = 1
+    POS_YAWRATE_CTRL = 2
+    VEL_CTRL = 3
+    ATT_CTRL = 4
+    RPYRT_CTRL = 5
+
+class Controller:
+    def __init__(self) -> None:
+        
+        # Required parameters
+        self.enable_debug = None
+        self.goal_pos = None
+
+        # Non-required parameters
+        self.is_sim = None
+        self.is_sim_default = False
+        self.loop_frequency = None
+        self.loop_frequency_default = 20
+        self.dist_thres = None
+        self.dist_thres_default = 0.1
+        
+
+        # ROS publishers and subscribers
+        self.state_sub = None
+        self.pos_pub = None
+        self.pos_yaw_pub = None
+        self.pos_yawrate_pub = None
+        self.vel_pub = None
+        self.att_pub = None
+        self.rpyrt_pub = None
+
+        # ROS services and clients
+        self.enable_control_srv = None
+        self.disable_control_srv = None
+        self.mission_finished_clt = None
+
+        # ROS message/service variables
+        self.pos_target_msg = PositionTarget()
+        self.att_target_msg = AttitudeTarget()
+        self.rpyrt_msg = RollPitchYawrateThrust()
+        self.cur_pose_msg = Pose()
+        self.cur_odom_msg = Odometry()
+        self.mission_finished_srv = Trigger()
+
+        # Convenience placeholder for converted messages
+        self.cur_pos = np.zeros((3,1))
+
+        # Controller variables
+        self.mission_finished = None
+        self.first_state_received = None
+        self.control_method = None
+
+        # Timer variables
+        self.print_timeout = None
+        self.timer_running = None
+        self.loop_timer = None
+        self.loop_start_time = None
+        self.last_time = None
+
+        self.first_reconfig_cb = None
+
+
+    def initialize(self) -> bool:
+        
+        # Initialize ROS interface
+        if not self._init_ros_interface():
+            rospy.logerr('[CONTROLLER]: Failed to initialize ROS interface')
```

Comment:
Can you do something similar for printing '[CONTROLLER]' in Python as what is done in C++ (helpers.h)? For example, now there is some inconsistency in printing in the function get_ros_parameters().